File: frontend/src/api/cartService.js
import axiosClient from './axiosClient';

/**
 * Shopping Cart API Service
 */

export const getCart = async () => {
  const response = await axiosClient.get('/api/cart');
  return response.data;
};

export const addToCart = async (productId, quantity = 1) => {
  const response = await axiosClient.post('/api/cart/items', {
    productId,
    quantity,
  });
  return response.data;
};

export const updateCartItem = async (itemId, quantity) => {
  const response = await axiosClient.put(`/api/cart/items/${itemId}`, {
    quantity,
  });
  return response.data;
};

export const removeCartItem = async (itemId) => {
  const response = await axiosClient.delete(`/api/cart/items/${itemId}`);
  return response.data;
};

export const clearCart = async () => {
  const response = await axiosClient.delete('/api/cart');
  return response.data;
};

File: mobile/lib/services/cart_service.dart
import 'package:dio/dio.dart';
import '../core/constants/api_constants.dart';
import '../core/network/dio_client.dart';
import '../models/cart_model.dart';

class CartService {
  final Dio _dio = DioClient().dio;

  /// Gets current user's shopping cart
  Future<CartModel> getCart() async {
    final response = await _dio.get(ApiConstants.cart);
    return CartModel.fromJson(response.data as Map<String, dynamic>);
  }

  /// Adds an item to the shopping cart
  Future<CartModel> addToCart({required int productId, int quantity = 1}) async {
    final response = await _dio.post(
      ApiConstants.cartItems,
      data: {
        'productId': productId,
        'quantity': quantity,
      },
    );
    return CartModel.fromJson(response.data as Map<String, dynamic>);
  }

  /// Updates quantity of an item in the cart
  Future<CartModel> updateItemQuantity({required int itemId, required int quantity}) async {
    final response = await _dio.put(
      '${ApiConstants.cartItems}/$itemId',
      data: {
        'quantity': quantity,
      },
    );
    return CartModel.fromJson(response.data as Map<String, dynamic>);
  }

  /// Removes an item from the cart
  Future<CartModel> removeItem(int itemId) async {
    final response = await _dio.delete('${ApiConstants.cartItems}/$itemId');
    return CartModel.fromJson(response.data as Map<String, dynamic>);
  }

  /// Clears the entire cart
  Future<CartModel> clearCart() async {
    final response = await _dio.delete(ApiConstants.cart);
    return CartModel.fromJson(response.data as Map<String, dynamic>);
  }
}
